mesonbuild/interpreterbase/interpreterbase.py

```diff
@@ -435,12 +471,15 @@ def evaluate_plusassign(self, node: mparser.PlusAssignmentNode) -> None:
     def evaluate_indexing(self, node: mparser.IndexNode) -> InterpreterObject:
         assert isinstance(node, mparser.IndexNode)
         iobject = self.evaluate_statement(node.iobject)
+        if iobject is None:
+            raise InvalidCodeOnVoid()
```

Comment:
AFAICT this is always safe, because of the:
```
        if iobject is None:
            raise InterpreterException('Tried to evaluate indexing on None')
```

Which you removed in this refactor. It's a bit better than "perform operation".

Also apparently we do, sometimes, refer to None objects, huh...